components/create_post/create_post.jsx

```diff
@@ -281,6 +281,8 @@ class CreatePost extends React.PureComponent {
     static getDerivedStateFromProps(props, state) {
         let updatedState = {currentChannel: props.currentChannel};
         if (props.currentChannel.id !== state.currentChannel.id) {
+            props.actions.setShowPreview(false);
```

Comment:
This should happen in `componentDidUpdate`. Actions shouldn't be called in `getDerivedStateFromProps`